client.go

```diff
@@ -689,19 +692,26 @@ func newLink(s *Session, r *Receiver, opts []LinkOption) (*link, error) {
 		MaxMessageSize:     l.maxMessageSize,
 	}
 
+	// copy source and target
+	if l.source != nil {
+		attach.Source = &(*l.source)
+	}
+	if l.target != nil {
+		attach.Target = &(*l.target)
+	}
```

Comment:
I don't think these lines are actually copying the the source/target (https://play.golang.org/p/5hG00uXingJ).

Is there a reason that they need to be copied? I don't see an obvious data race.

Reply:
Fixed.
Seems like go compiler just treats `&(*a)` as `a`, anyway I just didn't want `l.target` and `l.source` to be modified later on, seems like I just complicated things for nothing.